### PR TITLE
Adding auth

### PR DESCRIPTION
**Pull request by zoef-yt**
In this pr, I have added 
[Preview Link]()

- [x] Modal
- [x] Auth form
- [x] Auth logic with mockbee
- [x] Error handling
- [x] Login, SignUp and logout

Things not to check

- [ ] profile page
- [ ] Image is for representation purpose only will be changed
- [ ] profile page

### CODE REVIEW

File: src/components/modal/LoginCardModal.jsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		confirmPassword: '',
+		emailError: false,
+		passWordError: false,
+		confirmPasswordError: false,
+	};
+
+	const [textFields, setTextFields] = useState(defaultText);
+
+	const setUserDetails = (e) => {
+		console.log(e.target.name);
```
Comment (zoyatanweer) [inline src/components/modal/LoginCardModal.jsx]: remove console.log maybe?

File: src/components/header/header.jsx [hunk 2]
```diff
@@ -21,18 +26,46 @@ const Header = () => {
...
+							{totalCartQuantity > 0 ? <div className='badge-icon'>{totalCartQuantity > 9 ? '9+' : totalCartQuantity}</div> : <></>}
+						</NavLink>
+					</>
+				)}
 				<div onClick={toggleTheme}>{theme === 'dark' ? <SunIcon className='header-icon' /> : <HalfMoonIcon className='header-icon' />}</div>
-				<button className='btn btn-link'>Hello, Login</button>
+				{user != null ? (
+					<div className='badge-holder'>
+						<img
+							className='avatar avatar-sm'
```
Comment (santhoshpemmaka) [inline src/components/header/header.jsx:51]: Suggestion if possible to add alt tag in the image field.

File: src/components/modal/LoginCardModal.jsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+			: (setTextFields({ ...textFields, emailError: true }), errorHandler(true, 'Email is required'));
+	};
+
+	return (
+		<div className='modal-card'>
+			<div className='flex-column align-items-center space-evenly'>
+				<h1>{isLogin ? 'Login' : 'Sign Up'}</h1>
+				<div className='input-group'>
+					<InputField
+						labelText='Email Address'
```
Comment (santhoshpemmaka) [inline src/components/modal/LoginCardModal.jsx:79]: Suggestion if possible to add the required tag in the inputField for login and password.

**Review by santhoshpemmaka — APPROVED** [review]

File: src/components/css/modal.css [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	width: 100%;
+	z-index: 3;
+}
+.modal-error {
+	color: var(--colour-error);
+	font-size: 1.6rem;
+	margin: 0.5rem 0;
+	width: 30rem;
+	text-align: center;
+}
```
Comment (iRohitGaur) [inline src/components/css/modal.css:49]: suggestion: keep consistency in class spacing

**Comment by santhoshpemmaka** [comment]
Small suggestion my side this PR. Apart from code looks good and clean 💯 . Good to Merge 👍

**Review by iRohitGaur — APPROVED** [review]
UI looks great. Nice work. Good to merge 👍🏻

File: src/components/css/modal.css (hunk 4 above)
Comment (zoef-yt) [reply src/components/css/modal.css:49]: Yep, will do that 😂

File: src/components/header/header.jsx (hunk 2 above)
Comment (zoef-yt) [reply src/components/header/header.jsx:51]: Yep will add thanks

File: src/components/modal/LoginCardModal.jsx (hunk 1 above)
Comment (zoef-yt) [reply src/components/modal/LoginCardModal.jsx]: Thanks forgot to remove it

File: src/components/modal/LoginCardModal.jsx (hunk 3 above)
Comment (zoef-yt) [reply src/components/modal/LoginCardModal.jsx:79]: I've already added logic to check all the possibilities of input.

File: src/components/header/header.jsx [hunk 5]
```diff
@@ -1,16 +1,21 @@
+//@ts-check
 import React from 'react';
 import { ShoppingCartIcon, HeartIcon, SunIcon, HalfMoonIcon } from '../../assets/svg/svg';
 import { useTheme } from '../../context/theme/theme-context';
 import { Link, NavLink } from 'react-router-dom';
 import '../css/header.css';
 import { useWishlist } from '../../context/wishlist/wishlist-context';
 import { useMyCart } from '../../context/mycart/mycart-context';
+import { useModal } from '../../context/modal/modal-context';
+import { useAuth } from '../../context/auth/auth-context';
```
Comment (sudipkundu999) [inline src/components/header/header.jsx:10]: Suggestion : To keep an index.js in context folder,
then these 4 lines will reduce to import {useWishlist, useMyCart, useModal, useAuth} from "../../context"

**Review by sudipkundu999 — APPROVED** [review]
UI looks good. The functionality works perfectly. Good work!

**Review by zoyatanweer — APPROVED** [review]
clean code
easily understandable
good to go!

File: src/components/header/header.jsx (hunk 5 above)
Comment (zoef-yt) [reply src/components/header/header.jsx:10]: I will be doing it while refactoring the code thanks for the suggestion

**Pull request merged**